readthedocs/proxito/middleware.py

```diff
@@ -289,11 +289,24 @@ def process_request(self, request):  # noqa
         return None
 
     def add_hosting_integrations_headers(self, request, response):
+        addons = False
         project_slug = getattr(request, "path_project_slug", "")
+        version_slug = getattr(request, "path_version_slug", "")
+
         if project_slug:
             project = Project.objects.get(slug=project_slug)
+
+            # Check for the feature flag
             if project.has_feature(Feature.HOSTING_INTEGRATIONS):
-                response["X-RTD-Hosting-Integrations"] = "true"
+                addons = True
+
+            # Check if the version forces injecting the addons (e.g. using `build.commands`)
+            version = project.versions.filter(slug=version_slug).first()
+            if version and version.addons:
+                addons = True
```

Comment:
You can put this under an else, so it's run only the project didn't have the feature. And since you only need one attribute, you can use `values_list`.